com.unity.netcode.adapter.utp/Runtime/UTPTransport.cs

```diff
@@ -0,0 +1,835 @@
+#endif
+
+        /// <summary>
+        /// SendQueue dictionary is used to batch events instead of sending them immediately.
+        /// </summary>
+        private readonly Dictionary<SendTarget, SendQueue> m_SendQueue = new Dictionary<SendTarget, SendQueue>();
+
+        private void InitDriver()
+        {
+            DriverConstructor.CreateDriver(this, out m_Driver, out m_UnreliableSequencedPipeline, out m_ReliableSequencedPipeline, out m_ReliableSequencedFragmentedPipeline);
+        }
+
+        private void DisposeDriver()
+        {
+            if (m_Driver.IsCreated)
+            {
+                m_Driver.Dispose();
+            }
+        }
+
+        private NetworkPipeline SelectSendPipeline(NetworkDelivery delivery, int size)
+        {
+            switch (delivery)
+            {
+                case NetworkDelivery.Unreliable:
+                    return NetworkPipeline.Null;
+
+                case NetworkDelivery.UnreliableSequenced:
+                    return m_UnreliableSequencedPipeline;
+
+                case NetworkDelivery.Reliable:
+                case NetworkDelivery.ReliableSequenced:
+                    return m_ReliableSequencedPipeline;
+
+                case NetworkDelivery.ReliableFragmentedSequenced:
+                    // No need to send on the fragmented pipeline if data is smaller than MTU.
+                    if (size < NetworkParameterConstants.MTU)
+                    {
+                        return m_ReliableSequencedPipeline;
+                    }
+
+                    return m_ReliableSequencedFragmentedPipeline;
+
+                default:
+                    Debug.LogError($"Unknown NetworkDelivery value: {delivery}");
+                    return NetworkPipeline.Null;
+            }
+        }
+
+        private IEnumerator ClientBindAndConnect(SocketTask task)
+        {
+            var serverEndpoint = default(NetworkEndPoint);
+
+            if (m_ProtocolType == ProtocolType.RelayUnityTransport)
+            {
+                //This comparison is currently slow since RelayServerData does not implement a custom comparison operator that doesn't use
+                //reflection, but this does not live in the context of a performance-critical loop, it runs once at initial connection time.
+                if (m_RelayServerData.Equals(default(RelayServerData)))
+                {
+                    Debug.LogError("You must call SetRelayServerData() at least once before calling StartRelayServer.");
+                    task.IsDone = true;
+                    task.Success = false;
+                    yield break;
+                }
+
+                m_NetworkParameters.Add(new RelayNetworkParameter { ServerData = m_RelayServerData });
+            }
+            else
+            {
+                serverEndpoint = NetworkEndPoint.Parse(m_ServerAddress, m_ServerPort);
+            }
+
+            InitDriver();
+
+            if (m_Driver.Bind(NetworkEndPoint.AnyIpv4) != 0)
```

Comment:
Is IPv6 supported by the adapter? If so, might want to choose between this and `AnyIpv6` depending on the address family of `serverEndpoint`.

Alternatively, I don't think there's any need to bind on the client side. Could remove this entirely.

Reply:
We do support IPv6 in the transport but I don't think we support it currently in the adapter. As for binding I do think we need to do this for Relay since we overload the binding for making sure we bind for relay.

Reply:
I will add a ticket to support IPV6 to the transport API in general.